aspnetcore/signalr/scale.md

```diff
@@ -104,14 +104,70 @@ The preceding conditions make it likely to hit the 10 connection limit on a clie
 
 ## Linux with Nginx
 
-Set the proxy's `Connection` and `Upgrade` headers to the following for SignalR WebSockets:
+The following contains the minimum required settings to enable WebSockets, ServerSentEvents, and LongPolling for SignalR:
 
 ```nginx
-proxy_set_header Upgrade $http_upgrade;
-proxy_set_header Connection $connection_upgrade;
+http {
+  map $http_upgrade $connection_upgrade {
+    default Upgrade;
+    '' close;
+  }
```

Comment:
What's going on here?

Reply:
Like I was saying on the issue, "Connection: close" is applied by default in Nginx, and "Connection: keep-alive" should not be forwarded by proxies as it is a hop-by-hop header. We can have another discussion about that behavior, but I think for this PR this is a big improvement over what we had before.

Reply:
Now I remember. I have the same question [I had on the issue](https://github.com/dotnet/AspNetCore.Docs/issues/10827#issuecomment-631177400).

> Does this mean that every request to /hubroute will always contain either a "Connection: upgrade" or a "Connection: close" header? We wouldn't want to disable keep-alive for long polling.

Sure, "Connection" is a hop-by-hop header, but I want to make sure the connection between nginx and the signalr server isn't closed after every request.

Reply:
Is it safe to make all connections keep alive?

Reply:
I don't see why not. Most modern HTTP proxies and clients do. It can be problem with stateful NTLM connections, but we're not telling people how to configure NTLM.

@Tratcher

Reply:
Yes, enable keep-alive by default between nginx and kestrel.